lib/transaction.js

```diff
@@ -180,6 +188,12 @@ Transaction.prototype.prepareEnvironment = function() {
     return self.setIsolationLevel();
   }).then(function () {
     return self.setAutocommit();
+  }).catch(function (err) {
+    try {
+      self.rollback();
+    } catch (e) {}
+
+    throw err;
```

Comment:
Ah yes, that's actually more correct since we need to rethrow on a succesfull rollback aswell :)
Please put `finally` on the same line as `rollback` though.